src/main/java/jp/co/soramitsu/sora/crypto/service/impl/JSONVerifierImpl.java

```diff
@@ -0,0 +1,52 @@
+package jp.co.soramitsu.sora.crypto.service.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.IOException;
+import java.security.Signature;
+import java.security.SignatureException;
+import jp.co.soramitsu.sora.common.Util;
+import jp.co.soramitsu.sora.crypto.common.Consts;
+import jp.co.soramitsu.sora.crypto.proof.Options;
+import jp.co.soramitsu.sora.crypto.proof.Proof;
+import jp.co.soramitsu.sora.crypto.service.JSONCanonizer;
+import jp.co.soramitsu.sora.crypto.service.JSONVerifier;
+import lombok.AllArgsConstructor;
+
+
+@AllArgsConstructor
+public class JSONVerifierImpl implements JSONVerifier {
+
+  private final JSONCanonizer canonizer;
+  private final ObjectMapper mapper;
+
+  @Override
+  public boolean verify(Object object, Signature signature)
```

Comment:
I would recommend splitting the method on different private methods

Reply:
Sorry I do not understand. Like what methods? Any why private?

Reply:
I mean to refactor method so that it would more readable.
but I briefly tried to split it up... and now I think it is pretty clear :-)